src/compiler/checker.ts

```diff
@@ -48786,7 +48786,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         const parent = node.parent;
                         if (
                             node.kind === SyntaxKind.TypeParameter && !(isFunctionLikeDeclaration(parent) || isClassLike(parent) || isFunctionTypeNode(parent) ||
-                                isConstructorTypeNode(parent) || isCallSignatureDeclaration(parent) || isConstructSignatureDeclaration(parent) || isMethodSignature(parent))
+                                isConstructorTypeNode(parent) || isCallSignatureDeclaration(parent) || isConstructSignatureDeclaration(parent) || isMethodSignature(parent) || isJSDocTemplateTag(parent))

```

Comment:
typescript disallows `type X<const T> = [T]` but I don't think this disallows the equivalent

```js
/**
 * @template const T
 * @typedef {[T]} X
 */
```